### ISSUE TITLE
Decorator to be able to prepend and/or append to doc strings

### ISSUE THREAD

**Issue by nabobalis** [sunpy/sunpy#2377]
Once #2369 is merged in we have under `sunpy/sun/sun.py`, a decorator (called `add_parameter_docs`) that we use to append to the functions under `sun.py` as they all take the same parameter input.

We would like this to moved under to `sunpy/util/decorators.py` where we can import this function more generally as well as let it take arguments allowing us to pass in a prepend and/or append argument to the decorator that would add it to a decorated function's docstring. Maybe `.format` on the docstring if a dictionary is provided as an argument to the decorator.

Inspiration can come from https://github.com/pandas-dev/pandas/blob/master/pandas/core/generic.py#L477

**Comment by jnanjekye** [comment]
@Cadair , nabobalis. I would love to work on this.

**Comment by nabobalis** [comment]
@jnanjekye Thanks for the interest, unfortunately PR #2386 exists that tackles this issue. 

Feel free to tackle another issue and if you get stuck, send a message here and me or @Cadair should get in touch. Otherwise we have the [matrix channel.](https://riot.im/app/#/room/#sunpy:matrix.org)

**Comment by jnanjekye** [comment]
Noted.

**Comment by nabobalis** [comment]
@jnanjekye I would suggest having a look at the [Package Novice](https://github.com/sunpy/sunpy/labels/Package%20Novice) label. Those should be good starter issues to tackle, though some of them are not small fixes, the `Effort` label gives a rough indication of the scale of the problem.
